### ISSUE TITLE
update on `nixpkgs-channels` overwrites `nix-wrangle` pinning.

### ISSUE THREAD

**Issue by nathan-at-least** [timbertson/nix-wrangle#5]
**tl;dr:** Updating `nixpkgs` appears to overwrite the `nix-wrangle` dependency.

### Background

I'm trying out `nix-wrangle` for a new project and practicing some basic operations.

### Setup

First I created one git branch in my project where I run the following commands (with git commits after each one):

`nix-wrangle` branch:

- `nix-wrangle init --pkgs nixos-20.03`

`nix-wrangle-update-test` branch (starting from the same git parent as the `nix-wrangle` branch):

- `nix-wrangle init --pkgs nixos-19.09`
- `nix-wrangle update --type github --owner 'NixOS' --repo 'nixpkgs-channels' --ref 'nixos-20.03'`

The goal of my test was to observe if these two branches lead to identical working tree state.

### Bug Result

However, it looks like the second command on `nix-wrangle-update-test` branch does the wrong thing:

```
commit cf2851fe1dabfb453203eada66ade1ad3adc9b45 (HEAD -> nix-wrangle-update-test, local/nix-wrangle-update-test)
Author: Nate Wilcox <nathan+dev@electriccoin.co>
Date:   Sat Nov 21 10:22:19 2020 -0800

    Result of `nix-wrangle update --type github --owner 'NixOS' --repo 'nixpkgs-channels' --ref 'nixos-20.03'`

diff --git a/contrib/nix/wrangle.json b/contrib/nix/wrangle.json
index b91fb9c6f..e1a8adb6f 100644
--- a/contrib/nix/wrangle.json
+++ b/contrib/nix/wrangle.json
@@ -2,27 +2,27 @@
   "sources": {
     "nix-wrangle": {
       "fetch": {
-        "owner": "timbertson",
-        "repo": "nix-wrangle",
-        "rev": "9da4c0663d2578a3d7fe83e0795448da75d4e9c9",
-        "sha256": "08i4kdg9n2w5xmlnc52zz6mgshiax8725dlz3hf2v78d9aak0897"
+        "owner": "NixOS",
+        "repo": "nixpkgs-channels",
+        "rev": "f05c380a51daee53ac2edb0bac2fd5f1774e977a",
+        "sha256": "1xkgv4kvh2nii1kbxi0brjnb15scpzk8rkp7mzycgqh1lzfg23im"
       },
       "nix": "nix",
-      "owner": "timbertson",
-      "ref": "v1",
-      "repo": "nix-wrangle",
+      "owner": "NixOS",
+      "ref": "nixos-20.03",
+      "repo": "nixpkgs-channels",
       "type": "github"
     },
     "pkgs": {
       "fetch": {
         "owner": "NixOS",
         "repo": "nixpkgs-channels",
-        "rev": "75f4ba05c63be3f147bcc2f7bd4ba1f029cedcb1",
-        "sha256": "157c64220lf825ll4c0cxsdwg7cxqdx4z559fdp7kpz0g6p8fhhr"
+        "rev": "f05c380a51daee53ac2edb0bac2fd5f1774e977a",
+        "sha256": "1xkgv4kvh2nii1kbxi0brjnb15scpzk8rkp7mzycgqh1lzfg23im"
       },
       "nix": "default.nix",
       "owner": "NixOS",
-      "ref": "nixos-19.09",
+      "ref": "nixos-20.03",
       "repo": "nixpkgs-channels",
       "type": "github"
     }
```

### Expected behavior:

I expected the `nix-wrangle` package pinning info to remain unchanged while `nixpkgs-channel` pinning to be updated.

**Comment by nathan-at-least** [comment]
BTW- the diff between the two branches shows the unexpected delta. I expected this diff to be empty:

```
$ git diff nix-wrangle{,-update-test}                                                                                                          [2/1944]
diff --git a/contrib/nix/default.nix b/contrib/default.nix                                                                                             
similarity index 100%                                                                                                                                  
rename from contrib/nix/default.nix
rename to contrib/default.nix
diff --git a/contrib/nix/nix/wrangle.json b/contrib/nix/wrangle.json
similarity index 65%
rename from contrib/nix/nix/wrangle.json
rename to contrib/nix/wrangle.json
index 5aab2fbc8..e1a8adb6f 100644 
--- a/contrib/nix/nix/wrangle.json
+++ b/contrib/nix/wrangle.json
@@ -2,15 +2,15 @@
   "sources": {  
     "nix-wrangle": {
       "fetch": {
-        "owner": "timbertson",
-        "repo": "nix-wrangle",
-        "rev": "9da4c0663d2578a3d7fe83e0795448da75d4e9c9",
-        "sha256": "08i4kdg9n2w5xmlnc52zz6mgshiax8725dlz3hf2v78d9aak0897"
+        "owner": "NixOS",
+        "repo": "nixpkgs-channels",
+        "rev": "f05c380a51daee53ac2edb0bac2fd5f1774e977a",
+        "sha256": "1xkgv4kvh2nii1kbxi0brjnb15scpzk8rkp7mzycgqh1lzfg23im"
       },
       "nix": "nix",
-      "owner": "timbertson",
-      "ref": "v1",
-      "repo": "nix-wrangle",
+      "owner": "NixOS",
+      "ref": "nixos-20.03",
+      "repo": "nixpkgs-channels",
       "type": "github"
     },
     "pkgs": {
```

**Comment by timbertson** [comment]
Thanks for the report, I can see why this would happen. `update` can be used for two slightly different purposes, which is causing a problem here:

 - update a single package, passing e.g. a new git ref or version number explicitly on the commandline
 - updating multiple packages without further input, useful when you're tracking a git branch

What you want to do is specify that you're updating `nixpkgs`:

```
nix-wrangle update pkgs --ref 'nixos-20.03'
```

I think it makes sense to change `update` so that either you provide no arguments (to update dependencies tracking a git branch), or you _must_ provide a name, so that updating attributes (`--owner`, `--ref`, etc) only affects a single package.
